#!/usr/bin/env python
PACKAGE = "naze_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Velocity PID (uv)
gen.add("rollP", double_t, 0, "Proportional gain for roll", 4, 0, 8) # Default, Low, High
gen.add("rollI", double_t, 0, "Integral gain for roll", 0.03, 0, 1)
gen.add("rollD", double_t, 0, "Derivative gain for roll", 20, 0, 20)
gen.add("pitchP", double_t, 0, "Proportional gain for pitch", 4, 0, 8) # Default, Low, High
gen.add("pitchI", double_t, 0, "Integral gain for pitch", 0.03, 0, .1)
gen.add("pitchD", double_t, 0, "Derivative gain for pitch", 20, 0, 30)
gen.add("yawP", double_t, 0, "Proportional gain for yaw", 9, 0, 15) # Default, Low, High
gen.add("yawI", double_t, 0, "Integral gain for yaw", 0, 0, 1)
gen.add("yawD", double_t, 0, "Derivative gain for yaw", 0, 0, 20)

exit(gen.generate(PACKAGE, "naze_ros_node", "gains")) # Creates GainConfig.h (same name as file)



